### PR TITLE
Fix for hanging application, Submitting on behalf of Eric

### PR DESCRIPTION
**Pull request by nsarras**

### CODE REVIEW

File: dss_datamover/master_application.py [hunk 1]
```diff
@@ -204,7 +204,11 @@ def start(self):
         if not self.standalone:
             if not (self.operation.upper() == "LIST" and not self.config.get("distributed", False)):
                 self.stop_client_stale_process_on_clients()
-                self.spawn_clients()
+                if not self.spawn_clients():
```
Comment (nsarras) [inline dss_datamover/master_application.py:207]: this block executes if no clients were spun up. What about the case where only a fraction of the clients did not spawn? We want to be able to handle that scenario as well. We can compare the length of the successfully spawned clients with the # of expected clients. Then we can add logic to ensure that the application is not hanging waiting for those unspawned clients.

**Review by nsarras — COMMENTED** [review]
This is a good start at resolving connectivity issues. However, it only focuses on when we first spawn clients. It would be great to handle the situation where the connection is already established but we have not received any data or response from the clients for x amount of time, and then terminate/continue accordingly

File: dss_datamover/master_application.py (hunk 1 above)
Comment (nsarras) [reply dss_datamover/master_application.py:207]: Response From Author: "- I have tested this scenario, as I wrote in the validation part. The application did not hang itself to wait for unspawned clients, it only got stuck when all the clients failed in spawning. The application can function normally even if only one of the clients gets spawned successfully, so I only terminate it when no clients are spawned."

**Comment by nsarras** [comment]
> This is a good start at resolving connectivity issues. However, it only focuses on when we first spawn clients. It would be great to handle the situation where the connection is already established but we have not received any data or response from the clients for x amount of time, and then terminate/continue accordingly

Response from Author: "The no-response issue is indeed a problem. This patch only focuses on the spawn phase because this defect only mentions the passwordless access issue. I'll start working on it."

**Review by grandsuri — APPROVED** [review]
Looks good

**Pull request merged**